### ISSUE TITLE
The accuracy is not the same as what the paper shows

### ISSUE THREAD

**Issue by three-star-potato** [zjukg/KoPA#19]
![image](https://github.com/zjukg/KoPA/assets/59520602/1b5be3f4-f832-450c-a138-60f5219ab51c)
I just changed this line of code to make the code run, but the evaluation on UML was poor

![image](https://github.com/zjukg/KoPA/assets/59520602/008d9861-8781-480a-8191-25242559150a)
v2
![image](https://github.com/zjukg/KoPA/assets/59520602/e0de2f21-a7ef-42b9-b6cb-04a992856bc1)
Am I getting something wrong? Looking forward to your reply, thank you

**Comment by TAAANGY** [comment]
I dont change the code, the result was also poor
![image](https://github.com/zjukg/KoPA/assets/151164710/99a223d7-1553-4f7e-8af0-bede1401e818)

**Comment by Zhang-Each** [comment]
> ![image](https://private-user-images.githubusercontent.com/59520602/287969586-1b5be3f4-f832-450c-a138-60f5219ab51c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDgxODMsIm5iZiI6MTcwMTk0Nzg4MywicGF0aCI6Ii81OTUyMDYwMi8yODc5Njk1ODYtMWI1YmUzZjQtZjgzMi00NTBjLWExMzgtNjBmNTIxOWFiNTFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDExMTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4YTg5MmNkZTA5Njk0M2Y1MWQ1ZmY4YmRkZTdhYmY5NWYxNjk1OTdmZGU4ZTNmMTRjMjdjYmJmZWQxMDExMGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RwZhdoOaDHP3VHV6aPlw-23GUiD983E8ww_v_Habx7I) I just changed this line of code to make the code run, but the evaluation on UML was poor
> 
> ![image](https://private-user-images.githubusercontent.com/59520602/287970477-008d9861-8781-480a-8191-25242559150a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDgxODMsIm5iZiI6MTcwMTk0Nzg4MywicGF0aCI6Ii81OTUyMDYwMi8yODc5NzA0NzctMDA4ZDk4NjEtODc4MS00ODBhLTgxOTEtMjUyNDI1NTkxNTBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDExMTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiYjdkMmFjNTlmNDg0NjFhNGM3MmE0OTlhY2VmZTJlZTgxYTFmODQ3NTMzZjg3NDczZmQ0ODA5OTk4YTIyYTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NNzaLlSUrE67WD5nPLWh8pEPcTQwrc_ZbyY4tf9nTdY) v2 ![image](https://private-user-images.githubusercontent.com/59520602/287980890-e0de2f21-a7ef-42b9-b6cb-04a992856bc1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDgxODMsIm5iZiI6MTcwMTk0Nzg4MywicGF0aCI6Ii81OTUyMDYwMi8yODc5ODA4OTAtZTBkZTJmMjEtYTdlZi00MmI5LWI2Y2ItMDRhOTkyODU2YmMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDExMTgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwM2FlZThkZDFmZDdmODlkNjA2MmU5NzdkNjcyMTcxYmEyYTRjZDQ3ZWYzNzE2NWNjYTZhZjEzNTBmOTIxYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.I18xVqIGBEHiI35lf8N-FfY0scbfBWsL8OhW4-PdKqw) Am I getting something wrong? Looking forward to your reply, thank you

Hi, we think the possible reason is the mode crash due to floating point precision. We are using A800 GPU and CUDA 12.0 during the experiment. In this environment, the code can run normally and reproduce the corresponding results without any modification. If .half() is added it may cause some problems in mixed precision training. It's easy to have problems like yours with other kinds of GPUs and CUDA versions. Do you have any other questions?

**Comment by TAAANGY** [comment]
> > I dont change the code, the result was also poor ![image](https://private-user-images.githubusercontent.com/151164710/288356472-99a223d7-1553-4f7e-8af0-bede1401e818.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDgxODMsIm5iZiI6MTcwMTk0Nzg4MywicGF0aCI6Ii8xNTExNjQ3MTAvMjg4MzU2NDcyLTk5YTIyM2Q3LTE1NTMtNGY3ZS04YWYwLWJlZGUxNDAxZTgxOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwN1QxMTE4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMzRkMGFmZDEwYTk4NDM3NmMwNTZiYjg5ZDUyZTIwYThmZDc4Mzk3ZWUwZWI0MzI5NzQwODEzNmNjNmVkMWJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.I6yNgoaIwZlkuIArAXjm7MIOGipnwt9UNzCyzszzDfw)
> 
> Hello, have you run the code with the parameters in the README shell script? Because the default parameter settings in the finetune code are different from those in the shell script, if you run the finetune code directly instead of the shell script, the above situation may occur. According to our many attempts, the experimental results are reproducible. Do you have any other questions?

I run the shell script, as shown in the previous figure, the results are poor.

**Comment by Zhang-Each** [comment]
> > > I dont change the code, the result was also poor ![image](https://private-user-images.githubusercontent.com/151164710/288356472-99a223d7-1553-4f7e-8af0-bede1401e818.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDgxODMsIm5iZiI6MTcwMTk0Nzg4MywicGF0aCI6Ii8xNTExNjQ3MTAvMjg4MzU2NDcyLTk5YTIyM2Q3LTE1NTMtNGY3ZS04YWYwLWJlZGUxNDAxZTgxOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwN1QxMTE4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMzRkMGFmZDEwYTk4NDM3NmMwNTZiYjg5ZDUyZTIwYThmZDc4Mzk3ZWUwZWI0MzI5NzQwODEzNmNjNmVkMWJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.I6yNgoaIwZlkuIArAXjm7MIOGipnwt9UNzCyzszzDfw)
> > 
> > 
> > Hello, have you run the code with the parameters in the README shell script? Because the default parameter settings in the finetune code are different from those in the shell script, if you run the finetune code directly instead of the shell script, the above situation may occur. According to our many attempts, the experimental results are reproducible. Do you have any other questions?
> 
> I run the shell script, as shown in the previous figure, the results are poor.

You can try to train the model with UMLS-train-v2.json and tune the training epoch from 3 to 5 to get the best results. The number of negative samples in the data may affect the results to a great extent, and based on our attempts, the results in the paper are reproducible. We will also further optimize the code in the future, thanks for your attention.

**Comment by Zhang-Each** [comment]
> > > I dont change the code, the result was also poor ![image](https://private-user-images.githubusercontent.com/151164710/288356472-99a223d7-1553-4f7e-8af0-bede1401e818.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDgxODMsIm5iZiI6MTcwMTk0Nzg4MywicGF0aCI6Ii8xNTExNjQ3MTAvMjg4MzU2NDcyLTk5YTIyM2Q3LTE1NTMtNGY3ZS04YWYwLWJlZGUxNDAxZTgxOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwN1QxMTE4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMzRkMGFmZDEwYTk4NDM3NmMwNTZiYjg5ZDUyZTIwYThmZDc4Mzk3ZWUwZWI0MzI5NzQwODEzNmNjNmVkMWJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.I6yNgoaIwZlkuIArAXjm7MIOGipnwt9UNzCyzszzDfw)
> > 
> > 
> > Hello, have you run the code with the parameters in the README shell script? Because the default parameter settings in the finetune code are different from those in the shell script, if you run the finetune code directly instead of the shell script, the above situation may occur. According to our many attempts, the experimental results are reproducible. Do you have any other questions?
> 
> I run the shell script, as shown in the previous figure, the results are poor.

We apologize in the inconvenience caused to you, but we are sure that the results in the paper can be reproduced. This code is still in an early version, and we will further optimize the code and data in the future, thank you for your attention.
